src/intrinsics/ecma262/Object.js

```diff
@@ -180,6 +183,45 @@ export default function(realm: Realm): NativeFunctionValue {
       }
     }
     return to;
+  };
+  let ObjectAssign = func.defineNativeMethod("assign", 2, (context, args) => {
+    let effects;
+    try {
+      effects = realm.evaluateForEffects(() => {
+        return ObjectAssignInner(context, args);
+      });
+    } catch (e) {
+      if (realm.isInPureScope() && e instanceof FatalError) {
+        return realm.evaluateWithPossibleThrowCompletion(
+          () => AbstractValue.createTemporalFromBuildFunction(
+            realm,
+            ObjectValue,
+            [ObjectAssign, ...args],
+            ([methodNode, ..._args]: Array<BabelNodeExpression>) => {
+              return t.callExpression(methodNode, ((_args: any): Array<any>));
```

Comment:
Flow should know that __args is an Array<BabelNodeExpression> and that ought to be enough to type check the callExpression without needing the cast to any.

Reply:
If the first (to) argument is simple, the result will be simple as well. This might be useful to capture.

Reply:
I copied this from src/react/elements.js. callExpression is typed as `callExpression(callee: BabelNodeExpression, _arguments: Array<BabelNodeExpression | BabelNodeSpreadElement>): BabelNodeCallExpression` but AbstractValueBuildNodeFunction has Array<BabelNodeExpression> as argument. These aren't compatible.

If the flow types for t.callExpression are updated to take a $ReadOnlyArray then it doesn't need the cast. Not sure if we can just change `flow-typed`? I assume the changes are meant to be pushed upstream to flow-typed/too.

Reply:
Ah yes, co-variance lying in wait. It would be nice to get more accurate Flow types for Babel, but that is beyond the scope of this PR. For now, at least case to Array<BabelNodeExpression | BabelNodeSpreadElement> instead of Array<any>.